integrationtest/backend-tck/src/main/java/org/hibernate/search/v6poc/integrationtest/backend/tck/search/predicate/SearchPredicateIT.java

```diff
@@ -72,76 +65,53 @@ public void setup() {
-			}
-			catch (SearchException e) {
-				assertThat( e.getMessage() )
-						.contains( "Invalid value" )
-						.contains( "value to match" )
-						.contains( "must be non-null" )
-						.contains( fieldPath );
-			}
-		}
+		SearchQuery<DocumentReference> query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().match().onField( "string" ).matching( MATCHING_STRING )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, MATCHING_ID );
 	}
 
 	@Test
-	public void range_error_null() {
+	public void match_search_predicate() {
 		IndexSearchTarget searchTarget = indexManager.createSearchTarget().build();
 
-		List<String> fieldPaths = Arrays.asList(
-				"string", "string_analyzed", "integer", "localDate"
-		);
-
-		for ( String fieldPath : fieldPaths ) {
-			try {
-				searchTarget.predicate().range().onField( fieldPath ).from( null ).to( null );
-				fail( "Expected range() predicate with null bounds to throw exception on field " + fieldPath );
-			}
-			catch (SearchException e) {
-				assertThat( e.getMessage() )
-						.contains( "Invalid value" )
-						.contains( "at least one bound" )
-						.contains( "must be non-null" )
-						.contains( fieldPath );
-			}
-			try {
-				searchTarget.predicate().range().onField( fieldPath ).above( null );
-				fail( "Expected range() predicate with null bounds to throw exception on field " + fieldPath );
-			}
-			catch (SearchException e) {
-				assertThat( e.getMessage() )
-						.contains( "Invalid value" )
-						.contains( "at least one bound" )
-						.contains( "must be non-null" )
-						.contains( fieldPath );
-			}
-			try {
-				searchTarget.predicate().range().onField( fieldPath ).below( null );
-				fail( "Expected range() predicate with null bounds to throw exception on field " + fieldPath );
-			}
-			catch (SearchException e) {
-				assertThat( e.getMessage() )
-						.contains( "Invalid value" )
-						.contains( "at least one bound" )
-						.contains( "must be non-null" )
-						.contains( fieldPath );
-			}
-		}
+		SearchPredicate predicate = searchTarget.predicate().match().onField( "string" ).matching( MATCHING_STRING );
+
+		SearchQuery<DocumentReference> query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate( predicate )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, MATCHING_ID );
+	}
+
+	@Test
+	public void match_lambda() {
+		IndexSearchTarget searchTarget = indexManager.createSearchTarget().build();
+
+		SearchQuery<DocumentReference> query = searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate( c -> c.match().onField( "string" ).matching( MATCHING_STRING ) )
+				.build();
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, MATCHING_ID );
 	}
 
```

Comment:
Unless I'm wrong there are no tests for the `bool` predicate.
We can either add a separate class for it, or just add minimal tests here, since most of the tests will be about the different syntaxes (e.g. `BooleanJunctionPredicateContext#must(SearchPredicate)` vs. `BooleanJunctionPredicateContext#must()` vs. `BooleanJunctionPredicateContext#must(Consumer<? super SearchPredicateContainerContext<?>>)`).